### ISSUE TITLE
NoSuchFileException: /usr/local/zeebe/data/raft-partition/partitions/2/pending/...

### ISSUE THREAD

**Issue by pihme** [zeebe-io/zeebe#5499]
**Describe the bug**
Observed as error in production logs.

**Log/Stacktrace**
https://console.cloud.google.com/errors/CKiZxdXr5JmINA?service=zeebe&time=P7D&project=camunda-cloud-240911

**Environment:**
Camunda Cloud / Prod / 0.24.2

**Comment by deepthidevaki** [comment]
Happens when transitioning to a new role while receiving a snapshot. Transitioning is done first in Raft, while the the services for old role is still running in zeebe side. 

```
D 2020-11-12T11:35:13.140339Z Consume snapshot snapshotChunk 000652.sst of snapshot 164499-648-1605180910987-21475089655-21475089656 
D 2020-11-12T11:35:13.140480Z Waiting for more snapshot chunks of snapshot 164499-648-1605180910987-21475089655-21475089656, currently have 240/10275 
D 2020-11-12T11:35:13.161458Z Consume snapshot snapshotChunk 000654.sst of snapshot 164499-648-1605180910987-21475089655-21475089656 
D 2020-11-12T11:35:13.185194Z Deleted not completed (orphaned) snapshot /usr/local/zeebe/data/raft-partition/partitions/5/pending/164499-648-1605180910987-21475089655-21475089656-4 
E 2020-11-12T11:35:13.188357Z Unexpected error on writing the received snapshot chunk SnapshotChunkImpl{snapshotId=164499-648-1605180910987-21475089655-21475089656, totalCount=10275, chunkName='000654.sst', checksum=4113451613, snapshotChecksum=140918821} io.atomix.raft.snapshot.impl.SnapshotChunkImpl@711d3ae0 
D 2020-11-12T11:35:13.231289Z Partition role transitioning from FOLLOWER to CANDIDATE 
D 2020-11-12T11:35:13.806971Z Abort snapshot 164499-648-1605180910987-21475089655-21475089656 and mark it as invalid. 
D 2020-11-12T11:35:13.813247Z DELETE dir /usr/local/zeebe/data/raft-partition/partitions/5/pending/164499-648-1605180910987-21475089655-21475089656-4 
D 2020-11-12T11:35:13.827193Z Tried to delete pending dir /usr/local/zeebe/data/raft-partition/partitions/5/pending/164499-648-1605180910987-21475089655-21475089656-4, but doesn't exist. Either was already removed or no chunk was applied until now. 

```

Impact is that this node does not receive this snapshot. Otherwise broker continues to work with no problem.

**Comment by npepinpe** [comment]
Prioritized as mid, even though the impact is quite low, this is an issue that highlights the problems we have with our asynchronous transitions and how the distributed layer interacts with the application layer.

**Comment by npepinpe** [comment]
Looks like it also occurs when going the other way - see [here](https://console.cloud.google.com/errors/CKXP0fvJlryXMA?service=zeebe&time=P7D&project=camunda-cloud-240911). Looking at the logs, we see the node is leader and tries to commit a snapshot as it's transitioning to follower, which causes the pending snapshot to be deleted at the same time.

**Comment by korthout** [comment]
Encountered again for `0.25.2` on 12/6/20, 11:18 AM as [new error](https://console.cloud.google.com/errors/CNPFnb2e9NH2YQ?service=zeebe&time=P7D&project=camunda-cloud-240911&authuser=1). This appears to be the exact same situation, but with slight differences in its stacktrace.

**Comment by deepthidevaki** [comment]
This should be fixed by #6283
Please reopen if occured again.
